lib/rucio/core/rse.py

```diff
@@ -662,39 +658,51 @@ def get_rses_with_attribute_value(key, value, lookup_key, vo='def', session=None
 
 
 @read_session
-def get_rse_attribute(key, rse_id=None, value=None, use_cache=True, session=None):
+def get_rse_attribute_without_cache(rse_id: str, key: str, session: Optional[Session] = None) -> Optional[Union[str, bool]]:
     """
-    Retrieve RSE attribute value.
+    Retrieve RSE attribute value from the database.
 
     :param rse_id: The RSE id.
     :param key: The key for the attribute.
-    :param value: Optionally, the desired value for the attribute.
-    :param use_cache: Boolean to use memcached.
     :param session: The database session in use.
 
-    :returns: A list with RSE attribute values for a Key.
+    :returns: The value for the rse attribute, None if it does not exist.
     """
 
-    result = NO_VALUE
-    if use_cache:
-        result = REGION.get('%s-%s-%s' % (key, rse_id, value))
-    if result is NO_VALUE:
+    value = session.query(models.RSEAttrAssociation.value).filter_by(rse_id=rse_id, key=key).one_or_none()
 
-        rse_attrs = []
-        if rse_id:
-            query = session.query(models.RSEAttrAssociation.value).filter_by(rse_id=rse_id, key=key).distinct()
-            if value:
-                query = session.query(models.RSEAttrAssociation.value).filter_by(rse_id=rse_id, key=key, value=value).distinct()
-        else:
-            query = session.query(models.RSEAttrAssociation.value).filter_by(key=key).distinct()
-            if value:
-                query = session.query(models.RSEAttrAssociation.value).filter_by(key=key, value=value).distinct()
-        for attr_value in query:
-            rse_attrs.append(attr_value[0])
-        REGION.set('%s-%s-%s' % (key, rse_id, value), rse_attrs)
-        return rse_attrs
+    if value is None:
+        return None
 
-    return result
+    return value.value
+
+
+@read_session
+def get_rse_attribute(rse_id: str, key: str, session: Optional[Session] = None) -> Optional[Union[str, bool]]:
+    """
+    Retrieve RSE attribute value. If it is not cached, look it up in the
+    database. If the value exists and is not cached, it will be added to the
+    cache.
+
+    :param rse_id: The RSE id.
+    :param key: The key for the attribute.
+    :param session: The database session in use.
+
+    :returns: The value for the rse attribute, None if it does not exist.
+    """
+    cache_key = f'rse_attributes_{rse_id}_{key}'
+    value = REGION.get(cache_key)
+
+    if value is not NO_VALUE:
+        return value
+
+    value = get_rse_attribute_without_cache(rse_id, key, session=session)
+
+    if value is None:
+        return None
+
+    REGION.set(cache_key, value)
```

Comment:
Does it mean the missing attribute will never be cached ? seems like a possibility for performance issues.

Reply:
Yes, I'm thinking of the use case: add an RSE Attribute if it does not exist. This would set the cache to None for some time. In that time, the new RSE attribute can not be used.

Reply:
I will change the behavior in favor of performance, `None` values will be cached for now. The use case does not seem to appear that often.